javaparser-core/src/main/java/com/github/javaparser/Position.java

```diff
@@ -18,36 +18,82 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
  */
- 
+
 package com.github.javaparser;
 
 import com.github.javaparser.ast.Node;
 
-public class Position {
-    private int line;
-    private int column;
-
-    public static final Position ABSOLUTE_START = new Position(Node.ABSOLUTE_BEGIN_LINE,-1);
-    public static final Position ABSOLUTE_END = new Position(Node.ABSOLUTE_END_LINE,-1);
-
-    public static Position beginOf(Node node){
-        return new Position(node.getBeginLine(),node.getBeginColumn());
-    }
+/**
+ * A position in a source file. Lines and columns start counting at 1.
+ */
+public class Position implements Comparable<Position> {
+    public final int line;
+    public final int column;
 
-    public static Position endOf(Node node){
-        return new Position(node.getEndLine(),node.getEndColumn());
-    }
+    public static final Position ABSOLUTE_START = new Position(Node.ABSOLUTE_BEGIN_LINE, -1);
+    public static final Position ABSOLUTE_END = new Position(Node.ABSOLUTE_END_LINE, -1);
+    
+    /** The first position in the file */
+    public static final Position HOME = new Position(1, 1);
+    public static final Position UNKNOWN= new Position(0, 0);
```

Comment:
I agree, however these are the values that the code used to set if you set no position. 0,0,0,0. It's more about not wanting to break stuff than about having a good value. I can try setting them to -1,-1 and check what happens if you like?